consensus/ibft/extra.go

```diff
@@ -51,17 +41,19 @@ func PutIbftExtra(h *types.Header, istanbulExtra *IstanbulExtra) error {
 		extra = extra[:IstanbulExtraVanity]
 	}
 
-	data := istanbulExtra.MarshalRLPTo(nil)
-	extra = append(extra, data...)
-	h.ExtraData = extra
+	h.ExtraData = istanbulExtra.MarshalRLPTo(extra)
```

Comment:
Aren't you overriding the `extra` array here?

Why did you change this logic?

Reply:
Please check the implementation of `MarshalRLPTo`. It appends RLP data to given byte slice. extra is made expand or shrink before this in order to match fixed size slice.

Reply:
In previous code, there were 2 types code to set IBFT extra data to extra field. But I'm sure they are same.

Reply:
> Please check the implementation of MarshalRLPTo. It appends RLP data to given byte slice. extra is made expand or shrink before this in order to match fixed size slice.

I remember we had this exact same discussion a while back, I just wanted you to confirm it 🙂